### PR TITLE
PLNSRVCE-1469: add various performance diagnostic metrics to dashboard; fix alert panels not rendering by default

### PR DESCRIPTION
**Pull request by gabemontero**
This commit brings panels that display the various merics recently added to the exporter that measure various phases of a PipelineRuns execution, and renames "troubleshooting" to "performance analysis":
    - rename troubleshooting row and change threshold of successful pipelinerun from 80 to 600
    - add pipelinerun pod duration bargraph to analysis row
    - add kubelet mark schedulable bargraph to analysis row
    - add kubelet pulling images and start container bargraph to analysis row
    - rename avg sched duration to more precise learn of create duration
    - add pipelinerun pod avg duration by task, namespace bargraph to analysis row
    - move throttled by quota chart to analysis row
    - move throttled by node chart to analysis row
    - move throttled by quota to the left
    - add throttled by pvc graph to analysis row
    
This commit also hopefully finally addresses the alert panels not rendoring by default.

@openshift-pipelines/pipelines-service PTAL

tl;dr - I was seeing a lot of issues editing the panels with the grafana UI and exporting the JSON, and that process changing parts of the JSON not intended to be changed; this time, I copied panel json segments in jetbrains golang json editor, and then used gitops to push the change and verify in my `dev_setup.sh --use-local-branch` setup 
pipelines-performance-analysis-replace-trouble-shooting

### CODE REVIEW

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 1]
```diff
@@ -87,14 +87,11 @@
         "y": 0
       },
       "id": 93,
+      "repeat": "datasource",
       "title": "Alerts",
       "type": "row"
     },
     {
-      "datasource": {
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:94]: I believe this was part of the problem with the Alert panels not rendoring immediately

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 2]
```diff
@@ -147,10 +144,6 @@
       "pluginVersion": "9.1.6",
       "targets": [
         {
-          "datasource": {
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:150]: I believe this was part of the problem with the Alert panels not rendoring immediately

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 3]
```diff
@@ -167,10 +160,6 @@
       "type": "stat"
     },
     {
-      "datasource": {
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:170]: I believe this was part of the problem with the Alert panels not rendoring immediately

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 4]
```diff
@@ -222,10 +211,6 @@
       "pluginVersion": "9.1.6",
       "targets": [
         {
-          "datasource": {
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:225]: I believe this was part of the problem with the Alert panels not rendoring immediately

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 5]
```diff
@@ -87,14 +87,11 @@
         "y": 0
       },
       "id": 93,
+      "repeat": "datasource",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:90]: rather than those datasource ... ${DS_PROMETHEUS-APPSTUDIO-DS} entries, I saw this used in the rows/panels that "rendored immediately" and did not need me to initiate a manual "perform query" to fill out the alert panels.

Using this, I saw the alert panels rendor once the 30 minute window for our sample range passed in my dev_setup env

in stage/prod, since they are long running, we typically should have those metrics available from 30 minute ago and see as soon as we log in

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 6]
```diff
@@ -1067,85 +1052,6 @@
           "title": "Workqueue Depth",
           "type": "timeseries"
         },
-        {
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:1070]: moved these panels to the new performance analysis section

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 7]
```diff
@@ -1529,12 +1435,481 @@
...
+            }
+          ],
+          "yaxis": {
+            "align": false
+          }
         }
       ],
       "repeat": "datasource",
-      "title": "Pipelines Troubleshooting",
+      "title": "Pipelines Performance Analysis",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:1912]: changed the name of the row here

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 8]
```diff
@@ -1529,12 +1435,481 @@
               "refId": "A"
             }
           ],
-          "title": "Average scheduling duration",
+          "title": "Average time for learning of pipelinerun creation",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:1438]: this is where I start adding the new panels to the now named "peformance analysis" row

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 9]
```diff
@@ -87,14 +87,11 @@
...
       },
       "id": 93,
+      "repeat": "datasource",
       "title": "Alerts",
       "type": "row"
     },
     {
-      "datasource": {
-        "type": "prometheus",
-        "uid": "${DS_PROMETHEUS-APPSTUDIO-DS}"
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:96]: @Roming22  - we should in fact revert that change in your validation script, and no longer allow 

`"uid": "${DS_PROMETHEUS-APPSTUDIO-DS}"` 

as this at least with my dev_setup.sh testing was what caused the alert panels to not rendor automatically

Comment (gabemontero) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:96]: in fact @Roming22 we should be preventing any `"datasource":{...}` entries

Comment (gabemontero) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:96]: in favor of the ` "repeat": "datasource",` entry in the row the panel is a part of

**Review by ramessesii2 — APPROVED** [review]

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 9 above)
Comment (Roming22) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:96]: Sure. Let's connect to make sure I understand the requirements and make that happen.

**Review by Roming22 — APPROVED** [review]

**Comment by gabemontero** [comment]
Sigh @Roming22  ... something odd happened when I tried to rebase this PR after @ramessesii2 fix the the execute alert panel merged.

```
gmontero ~/go/src/github.com/openshift-pipelines/pipeline-service  (rework-troubleshooting)$ git push -uf origin rework-troubleshooting
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 12 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 1.93 KiB | 989.00 KiB/s, done.
Total 8 (delta 3), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.        
To github.com:gabemontero/pipeline-service.git
 ! [remote rejected] rework-troubleshooting -> rework-troubleshooting (cannot lock ref 'refs/heads/rework-troubleshooting': is at 4c09da68d41d008dfdc13f10cb5bc383870be700 but expected 610b83289632ff43e15b345a8772032bcba57552)
error: failed to push some refs to 'github.com:gabemontero/pipeline-service.git'

```

His merge commit is included in this PR now.  And the Rebase and merge button now fails for me.

Any idea @Roming22 what happened / is there a way to fix this?  Or do I need to create a new branch / PR?

**Comment by gabemontero** [comment]
replaced with https://github.com/openshift-pipelines/pipeline-service/pull/820
